src/Symfony/Component/Ldap/Adapter/ExtLdap/Query.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Ldap\Adapter\ExtLdap;
+
+use Doctrine\DBAL\Exception\DriverException;
+use Symfony\Component\Ldap\Adapter\AbstractQuery;
+use Symfony\Component\Ldap\Exception\LdapException;
+
+/**
+ * @author Charles Sarrazin <charles@sarraz.in>
+ */
+class Query extends AbstractQuery
+{
+    /** @var  Connection */
+    protected $connection;
+
+    /** @var resource */
+    private $search;
+
+    /**
+     * Constructor.
```

Comment:
The PHPDoc here overrides the class used for the Connection instance. This helps for the typehint, as we actually do not use the interface here (this class is specific to the implementation).

This is indeed needed for the `getResource()` method.